portfolio/src/main/webapp/script.js

```diff
@@ -38,12 +38,9 @@ function getData() {
 
     const dataListElement = document.getElementById('data-container');
     dataListElement.innerHTML = '';
-    dataListElement.appendChild(
-        createListElement('Comment 1: ' + data[0]));
-    dataListElement.appendChild(
-        createListElement('Comment 2: ' + data[1]));
-    dataListElement.appendChild(
-        createListElement('Comment 3: ' + data[2]));
+    data.forEach((item, index) => {
+      dataListElement.appendChild(createListElement('Comment ' + (index+1) + ': ' + item));
```

Comment:
Use String.format here to create a single string with the desired variable interpolation. Using '+' creates a new string each time with every new addition. This causes a lot of processing. 

https://stackoverflow.com/questions/6389827/string-variable-interpolation-java#6389845

Reply:
Not java my bad, use javascript string interpolation instead:
https://dmitripavlutin.com/string-interpolation-in-javascript/